### PR TITLE
mse

### PR DESCRIPTION
**Pull request by codingsneha**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,11 +5,13 @@ images/
...
 **/*.h5
 imuerror.json
 gt_speed.json
 imuestimate.json
 estimated_speed_mmPhase.json
 estimated_speed_doppler.json
-final_result.json
\ No newline at end of file
+final_result.json
+tempCodeRunnerFile.py
```
Comment (arghasen10) [inline .gitignore]: Dont add this in .gitignore. Simply delete this file.

File: create_merged_data.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+velocities_array = np.array(velocities)
+L_R_array = np.array(L_R)
+
+data_dict = {'rangeHeatmap': rangeHeatmaps_array, 'velocity': velocities_array, 'L_R': L_R_array}
+
+with open('merged_data.pkl', 'wb') as f:
+    pickle.dump(data_dict, f)
+
+print(f"Merged data saved to merged_data.pkl")
+# print(rangeHeatmaps.shape)
```
Comment (arghasen10) [inline create_merged_data.py]: Remove unnecessary comments from line 67:76

File: helper.py [hunk 3]
```diff
@@ -350,6 +355,7 @@ def solve_equation(phase_cur_frame,info_dict):
     for root in roots_of_frame:
         if root >0.9*median_root and root<1.1*median_root:
             final_roots.append(root)
+        print(root)
```
Comment (arghasen10) [inline helper.py]: Dont keep print statement.

File: helper.py [hunk 4]
```diff
@@ -490,3 +555,206 @@ def get_mode_velocity(velocity_array_framewise):
...
+    noiserp_train = np.array(
+        [np.concatenate([np.expand_dims(e.reshape(-1, 1), 2) for e in v.transpose(1, 0)[1]], axis=2) for v in X_train])
+
+    dop_train_s = (dop_train - dop_train.min()) / (dop_train.max() - dop_train.min())
+    rp_train_s = (rp_train - rp_train.min()) / (rp_train.max() - rp_train.min())
+    noiserp_train_s = (noiserp_train - noiserp_train.min()) / (noiserp_train.max() - noiserp_train.min())
+    return dop_train_s, rp_train_s, noiserp_train_s
+
+
+# def preprocess_input(X_train):
```
Comment (arghasen10) [inline helper.py]: Remove unwanted comments 703:729

File: import_all.py [hunk 5]
```diff
@@ -11,7 +11,6 @@
 import seaborn as sns
 import argparse
 import pandas as pd
-import cv2
```
Comment (arghasen10) [inline import_all.py]: Dont delete this.

File: main.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+from helper import *
```
Comment (arghasen10) [inline main.py]: We don't need this file in this PR.

File: requirements.txt [hunk 7]
```diff
@@ -0,0 +1,5 @@
+matplotlib==3.9.0
```
Comment (arghasen10) [inline requirements.txt]: Dont add this. This is not required for the PR.

**Review by arghasen10 — CHANGES_REQUESTED** [review]
Make all the requested changes to update the PR.

**Comment by codingsneha** [comment]
> Make all the requested changes to update the PR.

Done

File: helper.py [hunk 8]
```diff
@@ -331,8 +334,10 @@ def solve_equation(phase_cur_frame,info_dict):
         phase_diff.append(phase_cur_frame[soham]-phase_cur_frame[soham-1])
     Tp=cfg.Tp
     Tc=cfg.Tc
-    L=info_dict[' L'][0]/100
-    r0=info_dict[' R'][0]/100
+    L = info_dict[0]/100
+    r0 = info_dict[1]/100
+    #L=info_dict[' L'][0]/100
+    #r0=info_dict[' R'][0]/100
```
Comment (arghasen10) [inline helper.py]: Why?

File: helper.py [hunk 9]
```diff
@@ -490,3 +554,178 @@ def get_mode_velocity(velocity_array_framewise):
...
+        self.i = self.i+1
+        pinn_loss = tf.reduce_mean(tf.square(calculated_value - y_pred))
+
+        # MSE loss calculation
+        mse_loss = tf.reduce_mean(tf.square(y_true - y_pred))
+
+        # Combined PINN loss and MSE loss
+        combined_loss = (1 - self.mse_weight) * pinn_loss + self.mse_weight * mse_loss
+
+        return combined_loss
```
Comment (arghasen10) [inline helper.py]: @codingsneha don't add anything related to PINN in this PR. 

Will merge PINN n the next PR.

File: helper.py [hunk 10]
```diff
@@ -490,3 +554,178 @@ def get_mode_velocity(velocity_array_framewise):
...
+# def get_cnn1d():
+#     model1d = tf.keras.Sequential([
+#         tf.keras.layers.Conv2D(32, (8, 2), (2, 1), padding="valid", activation='relu', input_shape=(64, 2, 10)),
+#         tf.keras.layers.Conv2D(64, (8, 1), (2, 1), padding="valid", activation='relu'),
+#         tf.keras.layers.Conv2D(96, (4, 1), (2, 1), padding="valid", activation='relu'),
+#         tf.keras.layers.GlobalAveragePooling2D(),
+#         tf.keras.layers.Dropout(rate=0.3)
+#     ], name='cnn1d')
+#     return model1d
+
```
Comment (arghasen10) [inline helper.py]: Dont add comments.

File: helper.py [hunk 11]
```diff
@@ -490,3 +554,178 @@ def get_mode_velocity(velocity_array_framewise):
...
+
+def get_model():
+    cnn = get_cnn()
+    # cnn1d = get_cnn1d()
+    input = tf.keras.layers.Input(shape=(182, 256, 1))
+    # input_2 = tf.keras.layers.Input(shape=(64, 1, 10))
+    # input_3 = tf.keras.layers.Input(shape=(64, 1, 10))
+    # input_23 = tf.keras.layers.Concatenate(axis=2)([input_2, input_3])
```
Comment (arghasen10) [inline helper.py]: Please remove all unnecessary comments.

File: helper.py [hunk 12]
```diff
@@ -490,3 +554,178 @@ def get_mode_velocity(velocity_array_framewise):
...
+            batch_size=32,
+        )
+    plt.plot(history.history['loss'], label='MSE Loss')
+    plt.plot(history.history['val_loss'], label='Validation MSE Loss')
+    plt.plot(history.history['SolveEquationLoss'], label='Combined Loss')
+    plt.plot(history.history['val_SolveEquationLoss'], label='Validation Combined Loss')
+    plt.legend()
+    plt.show()
+    return model
+
```
Comment (arghasen10) [inline helper.py]: Why 2 train functions. Add only 'mse' loss related train function in this PR.

File: test.py [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+
+for file_name in bin_files:
+    print("Processing file ", file_name)
+    info_dict = get_info(file_name)
+    run_data_read_only_sensor(info_dict)
+    bin_filename = 'datasets/only_sensor' + info_dict['filename'][0]
+    bin_reader = RawDataReader(bin_filename)
+    total_frame_number = int(info_dict[' Nf'][0])
+    pointCloudProcessCFG = PointCloudProcessCFG()
+    print_info(info_dict)
```
Comment (arghasen10) [inline test.py]: I think we don't need this file.

**Review by arghasen10 — CHANGES_REQUESTED** [review]
Almost there!!

@codingsneha  make the requested changes. We will integrate PINN in the next PR. So don't add those changes to this PR #1  

And dont make the comments resolved. I will do that.

File: helper.py (hunk 8 above)
Comment (codingsneha) [reply helper.py]: It was required for PINN, passing the L_R array instead of info_dict from the merged_data L_R[0] gives L and L_R[1] gives R. Since it is not required for MSE I removed it in recent commit along with other PINN code.

File: helper.py (hunk 9 above)
Comment (codingsneha) [reply helper.py]: Got it, updated.

File: helper.py (hunk 10 above)
Comment (codingsneha) [reply helper.py]: Removed

File: helper.py (hunk 11 above)
Comment (codingsneha) [reply helper.py]: Done

File: helper.py (hunk 12 above)
Comment (codingsneha) [reply helper.py]: Removed PINN train function

File: test.py (hunk 13 above)
Comment (codingsneha) [reply test.py]: Removed sir

**Review by arghasen10 — APPROVED** [review]
LGTM.
